generators/arduino/matrix.js

```diff
@@ -141,18 +129,18 @@ Blockly.Arduino.oxocard_matrix_draw_circle = function() {
 
 Blockly.Arduino.oxocard_matrix_draw_number = function() {
 	var num = Blockly.Arduino.valueToCode(this, 'NUMBER', Blockly.Arduino.ORDER_NONE);
-	return 'oxocard.matrix->drawNumber(' +num +');\n';
+	return 'oxocard.matrix->drawNumber(' + num +');\n';
 };
 
 Blockly.Arduino.oxocard_matrix_draw_text = function() {
 	var text = Blockly.Arduino.valueToCode(this, 'TEXT', Blockly.Arduino.ORDER_NONE);
 	var isBigFont = this.getFieldValue('BUTTON').toLowerCase();
-	return 'oxocard.matrix->drawText(' +text +',' +isBigFont +');\n';
+	return 'oxocard.matrix->drawText(' + text + ', ' + isBigFont +');\n';
 };
 
 Blockly.Arduino.oxocard_matrix_draw_weather = function() {
-	return "oxocard.matrix->clear();\n"
-			+"oxocard.matrix->setForeColor(oxocard.weather->getDrawableIconColor());\n"
-			+"oxocard.matrix->drawImage(oxocard.weather->getDrawableIcon());\n"
-			+"oxocard.matrix->update();\n";
+	return 'rgbColor_t color = oxocard.weather->getDrawableIconColor();\n'
+		+ 'uint8_t *imagePtr = oxocard.weather->getDrawableIcon();\n'
```

Comment:
local variables should be kept at a very minimum. or at the very least use proper naming.

imagine someone in blockly using the variable "color" (which is _very_ realistic). You would break to whole thing...

also a good test is: what would happen if 2 people use the same block twice? in this case everything (again) gets broken.